app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,8 @@
+class PostsController < ApplicationController
+  def index
+    @user = User.find(params[:user_id])
+    @posts = @user.recent_posts
+  end
+
+  def show; end
```

Comment:
- [ ] In line with the above suggestion, this show action needs implementation for us to be able to display a single post. Please consider the suggestion below.
```suggestion
  def show
    @user = User.find(params[:user_id])
    @post = @user.posts.find(params[:id])
  end
```